backend/src/ds/self_remove.rs

```diff
@@ -60,10 +60,15 @@ impl DsGroupState {
         // Everything seems to be okay.
         // Now we have to update the group state and distribute.
 
+        let sender = VerifiableClientCredential::try_from(
+            self.group.leaf(sender_index).unwrap().credential().clone(),
+        )
+        .unwrap();
+
         if let Err(e) = self.room_state.apply_regular_proposals(
-            &sender_index.u32(),
+            sender.user_id(),
             &[MimiProposal::ChangeRole {
-                target: sender_index.u32(),
+                target: sender.user_id().clone(),
```

Comment:
The API of mimi-policy is a little bit unfortunate. Technically, it does not need ownership of `UserId` at all and also it does not know what a user id is. It operates on opaque bytes identifiers. So, just an idea: mimi-policy could have an Identifier type which just wraps bytes.

Let's not change it now. I just wanted to leave this comment here.